### PR TITLE
Add required fields to NewCaseForm

### PR DESCRIPTION
**Pull request by allysonjp715**
I removed two unit tests that checked that requests were sent because those requests are no longer sent until all the required fields are filled out. That unit test didn't provide any added coverage compared to the integration tests, so I don't think they are worth keeping around. Let me know if you disagree.

In future PRs we can look into using <FastField>s to speed up the I/O on the form: https://github.com/formik/formik/issues/671

### CODE REVIEW

File: verification/curator-service/ui/src/components/NewCaseForm.test.tsx [hunk 1]
```diff
@@ -27,38 +19,3 @@ it('renders form', () => {
...
-    const { getByText } = render(<NewCaseForm user={user} />);
-    expect(getByText(/Submit case/i)).toBeInTheDocument();
-
-    await wait(() => {
-        fireEvent.click(getByText(/Submit case/));
-    });
-    expect(mockedAxios.post).toHaveBeenCalledTimes(1);
-});
-
-it('submits case not ok', async () => {
```
Comment (attwad) [inline verification/curator-service/ui/src/components/NewCaseForm.test.tsx:50]: we should be testing for a POST vs PUT really in those tests depending on if we had an initial case or not, I thought I added that but no?

File: verification/curator-service/ui/src/components/new-case-form-fields/Events.tsx [hunk 2]
```diff
@@ -27,11 +27,13 @@ export default function Events(): JSX.Element {
                 <DateField
                     name="confirmedDate"
                     label="Confirmed case date"
+                    required={true}
```
Comment (attwad) [inline verification/curator-service/ui/src/components/new-case-form-fields/Events.tsx:30]: nit. can we just say `required` ? if no no big deal.

Comment (allysonjp715) [reply verification/curator-service/ui/src/components/new-case-form-fields/Events.tsx]: Done

File: verification/curator-service/ui/src/components/NewCaseForm.test.tsx (hunk 1 above)
Comment (allysonjp715) [reply verification/curator-service/ui/src/components/NewCaseForm.test.tsx:50]: Those checks aren't in this test file in the master branch.

We do check in cypress tests if it's a POST or PUT by which route we wait on, comparing NewCaseFormTest and EditCaseTest

**Comment by allysonjp715** [comment]
[Finish new case manual entry page](https://trello.com/c/haOgIswr/246-finish-new-case-manual-entry-page)

File: verification/curator-service/ui/src/components/NewCaseForm.test.tsx (hunk 1 above)
Comment (attwad) [reply verification/curator-service/ui/src/components/NewCaseForm.test.tsx:64]: ok nm then, thanks

**Pull request merged**